File: src/com/mojang/escape/level/block/ChestBlock.java
package com.mojang.escape.level.block;

import com.mojang.escape.*;
import com.mojang.escape.entities.Item;
import com.mojang.escape.gui.BasicSprite;
import com.mojang.escape.gui.Sprite;
import com.mojang.escape.level.Level;

public class ChestBlock extends Block {
	private boolean open = false;
	private Sprite chestSprite;

	public ChestBlock() {
		setTex(1);
		setBlocksMotion(true);

		chestSprite = new BasicSprite(0, 0, 0, 8 * 2 + 0, Art.INSTANCE.getCol(0xffff00));
		addSprite(chestSprite);
	}

	public boolean use(Level level, Item item) {
		if (open) return false;

		chestSprite.setTex(chestSprite.getTex() + 1);
		open = true;

		level.getLoot(getId());
		Sound.Companion.getTreasure().play();

		return true;
	}
}

File: src/com/mojang/escape/level/block/ChestBlock.kt
package com.mojang.escape.level.block

import com.mojang.escape.Art
import com.mojang.escape.Sound
import com.mojang.escape.entities.Item
import com.mojang.escape.gui.BasicSprite
import com.mojang.escape.level.Level

class ChestBlock: Block() {
    private val chestSprite = BasicSprite(0.0, 0.0, 0.0, 8 * 2 + 0, Art.getCol(0xffff00))
    private var open = false

    init {
        tex = 1
        blocksMotion = true

        addSprite(chestSprite)
    }

    override fun use(level: Level, item: Item): Boolean {
        if (open) {
            return false
        }

        chestSprite.tex = chestSprite.tex + 1
        open = true

        level.getLoot(id)
        Sound.treasure.play()

        return true
    }
}